trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/standardfunctions/StandardFunctionHandler.java

```diff
@@ -2421,12 +2419,12 @@ private Result handleErrorFunction(final IDispatcher main, final IASTFunctionCal
 	 * settings. If we want to check reachability, an assert false will be generated. If not, (e.g., if we only want to
 	 * check memsafety), an assume false will be generated.
 	 */
-	private Statement createReachabilityAssert(final ILocation loc, final String functionName) {
-		final boolean checkSvcompErrorfunction = mSettings.checkSvcompErrorFunction();
+	private Statement createReachabilityAssert(final ILocation loc, final String functionName,
+			final boolean checkProperty, final Spec spec) {
 		final boolean checkMemoryleakInMain = mSettings.checkMemoryLeakInMain()
 				&& mMemoryHandler.getRequiredMemoryModelFeatures().isMemoryModelInfrastructureRequired();
 		final Expression falseLiteral = ExpressionFactory.createBooleanLiteral(loc, false);
-		if (!checkSvcompErrorfunction && !checkMemoryleakInMain) {
+		if (!checkProperty && !checkMemoryleakInMain) {

```

Comment:
Does this mean that, if we don't check asserts, we assume they hold? If so, is that the right behaviour? And if so, shouldn't the same be done in `handleAssert` ?

Reply:
> Does this mean that, if we don't check asserts, we assume they hold? If so, is that the right behaviour?

Seems like this is the case. I did not introduce the support for this, but just adapted the settings to handle it separately. We could also use the three-value-enum (IGNORE, ASSUME, CHECK) here, if we want to be more flexible.

> And if so, shouldn't the same be done in handleAssert ?

Yes, you are right, `handleAssert` should be handled similarly.

Reply:
Please update the documentation comment for the method.